src/modules/mock/mock.js

```diff
@@ -0,0 +1,137 @@
+import { getRandomArrayElement, getRandomInteger } from '../../global/utils.js';
+
+const MIN_DEST_DESCR_RANDOM_VALUE = 1;
+const MAX_DEST_DESCR_RANDOM_VALUE = 5;
+const MIN_PICTURE_COUNT = 0;
+const MAX_PICTURE_COUNT = 5;
+const MIN_SRC_PICTURE_RANDOM_VALUE = 1;
+const MAX_SRC_PICTURE_RANDOM_VALUE = 9;
+const MIN_OFFER_COUNT = 0;
+const MAX_OFFER_COUNT = 8;
+const MIN_OFFER_PRICE_VALUE = 200;
+const MAX_OFFER_PRICE_VALUE = 2000;
+const MIN_EVENT_BASE_PRICE = 1000;
+const MAX_EVENT_BASE_PRICE = 5000;
+
+let destinationId = 0;
+let pictureId = 0;
+let eventTypeId = 0;
+let eventTypeOfferId = 0;
+let eventId = 0;
+
+const destinationDescriptions = [
+  'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget.',
+  'Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.',
+  'Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.',
+  'Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.',
+  'Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.',
+  'Sed sed nisi sed augue convallis suscipit in sed felis.',
+  'Aliquam erat volutpat.',
+  'Nunc fermentum tortor ac porta dapibus.',
+  'In rutrum ac purus sit amet tempus.'
+];
+
+const destinationNames = [
+  'Moscow',
+  'Paris',
+  'London',
+  'Pskov',
+  'Omsk'
+];
+
+const eventTypes = [
+  'taxi',
+  'bus',
+  'train',
+  'ship',
+  'drive',
+  'flight',
+  'check-in',
+  'sightseeing',
+  'restaurant'
+];
+
+//! Пункт назначения
+
+const createDestinationDescription = (descriptions) => {
+  const descriptionArray = Array.from({ length: getRandomInteger(MIN_DEST_DESCR_RANDOM_VALUE, MAX_DEST_DESCR_RANDOM_VALUE) }, () => getRandomArrayElement(descriptions));
+
+  return [...new Set(descriptionArray)].join(' ');
+};
+
+const createPictureData = () => (
+  {
+    src: `https://loremflickr.com/248/152?random=${getRandomInteger(MIN_SRC_PICTURE_RANDOM_VALUE, MAX_SRC_PICTURE_RANDOM_VALUE)}`,
+    description: `Some picture description #${++pictureId}`
+  }
+);
+
+const createPictureDataset = () => Array.from({ length: getRandomInteger(MIN_PICTURE_COUNT, MAX_PICTURE_COUNT) }, createPictureData);
+
+const createDestinationData = () => {
+  pictureId = 0;
+
+  return {
+    id: ++destinationId,
+    description: createDestinationDescription(destinationDescriptions),
+    name: destinationNames[destinationId - 1],
+    pictures: createPictureDataset()
+  };
+};
+
+const createDestinationDataset = () => Array.from({ length: destinationNames.length }, createDestinationData);
+
+//! Опции
+
+const createOfferData = () => (
+  {
+    id: ++eventTypeOfferId,
+    title: `Some offer #${eventTypeOfferId}`,
+    price: getRandomInteger(MIN_OFFER_PRICE_VALUE, MAX_OFFER_PRICE_VALUE)
+  }
+);
+
+const createOfferDataset = () => Array.from({ length: getRandomInteger(MIN_OFFER_COUNT, MAX_OFFER_COUNT) }, createOfferData);
```

Comment:
Можно не заводить лишние переменные, а прокидывать индекс в функцию отсюда
```suggestion
const createOfferDataset = () => Array.from({ length: getRandomInteger(MIN_OFFER_COUNT, MAX_OFFER_COUNT) }, (_, i) => createOfferData(i));
```